plugins/readtimer.koplugin/main.lua

```diff
@@ -187,4 +193,13 @@ function ReadTimer:addToMainMenu(menu_items)
     }
 end
 
+-- The UI ticks on a monotonic time domain, while this plugin deals with real time.
+-- Make sure we fire the alarm right away if it expired during suspend...
+function ReadTimer:onResume()
+    if self:remainingMinutes() == 0 then
+        self:alarm_callback()
+        self:unschedule()
+    end
```

Comment:
Actually, if larger than 0 then unschedule and reschedule subtracted version? :angel:

Reply:
It's already effectively the subtracted version by virtue of being scheduled on the MONOTONIC timescale (i.e., it's scheduled in n seconds in the future, and those seconds only tick while awake) ;).

Reply:
By subtracted I mean:

1. You set timer for 60 minutes.
2. At some point in, you sleep for 20 minutes.
3. You want the timer to go off at 60 minutes. *Not*, very, very, very much not at 60+20, and even less not at all.

So subtracted would be remaining - 20.

Reply:
There's going to be precision loss because we're crossing time domains, and we'd essentially be checking against what the timer's expiration date *should* theoretically be in one domain (REAL) while it's actually scheduled and ticking in another (MONO) ;).

(Given the UI domain, race conditions are less likely this way around, though).

We don't actually have a way to query the current timer for a specific task in its native domain (i.e., MONO) and we also don't have a centralized way of computing the sleep duration (some of our target devices are too old to support BOOTTIME), and I'm not aware of any sane way to get at the accurate computation of this delay the kernel does as part of its timekeeping duties...

TL;DR: The only way I can see this being done is simply to compare os.time() from suspend & resume. If it drifts or shit happened, it's going to be even more broken ^^.